### ISSUE TITLE
Extremely slow converting to avif

### ISSUE THREAD

**Issue by negezor** [image-rs/image#1760]
This happens in any image, example https://imgur.com/a/5YNiz89

## Expected

Quickly converted image to avif

## Actual behaviour

More than a minute and a half conversion time for an example image. In this case, only one core is used. At the same time, converting to other types of images occurs instantly.

```sh
cargo run --bin avif_repro --release
    Finished release [optimized] target(s) in 0.16s
     Running `target/release/avif_repro`
project master took 1m56s
```

## Reproduction steps

Run with `--release`
```rs
fn main() {
    let im = image::open("fox.png").unwrap();

    im.save("fox.avif").unwrap();
}
```

## Additional info
- OS: `WSL Arch Linux`
- Kernel: `5.10.60.1-microsoft-standard-WSL2`
- CPU: `AMD Ryzen 9 3900X`
- Rust: `1.61.0`
- Crate version: `0.24.2`
- Enabled crate features: `["png", "avif"]`

If you need information that I can provide, just let me know.

**Comment by fintelia** [comment]
I fear this may just be a consequence of avif being slower to encode than other formats. As you note, it should be possible to parallelize the encoding process (PRs welcome!), but beyond that I'm not sure whether there's really much more low hanging fruit to improve things

**Comment by mothsART** [comment]
![compare_image-rs--vs--imagemagick](https://user-images.githubusercontent.com/10601196/185371147-1130850f-8de6-4acd-bfda-d7227c3d456c.png)

I think, encoding algorithm must be compare to imagemagick. More than 400x slower and same checksum at the end.

**Comment by anarchodin** [comment]
> I fear this may just be a consequence of avif being much slower to encode than other formats.

I just ran a test on an image I have. I killed the binary using this crate after more than seven minutes with no results. That same photo recodes in [Squoosh](https://squoosh.app) in 2.7 seconds on the default settings. Clearly not something inherent to AVIF.

Which got me thinking - there's absolutely no access to configuration options with the simple API, so just what _are_ the options being passed to the encoder?

Turns out, it's being [told](https://github.com/image-rs/image/blob/b02bb7426cc716d9353fac19ac1d744c8b298f41/src/codecs/avif/encoder.rs#L59) to use the _slowest_ speed configuration available, with the _highest_ quality configuration. This for a codec which infamously _aspires_ to reaching glacial speeds. I figured out how to encode an image with some saner settings using `image::codecs::avif::AvifEncoder` (no trace of which exists in the documentation, incidentally). Using those, I do get an image in less than a minute, and even heading towards useful timeframes at the fastest settings. There seems to be some parallelisation involved here: My test shell isn't doing anything to cause threading, but there are four active threads while it's running.

It's still _clearly_ slower than running the file through Squoosh, but not two orders of magnitude slower. Considering that Squoosh is using a wasm version of ` libaom`, and `rav1e` is specifically claimed to be good for use cases where `libaom` is too slow, I find this strange, to say the least.

@fintelia: It might be a good idea to change that default, and possibly make it easier to tweak the settings.
